packages/core/src/ui/InlineToolbar/index.ts

```diff
@@ -1,204 +1,182 @@
-      }
-    });
+    this.#nodes.holder.appendChild(this.#nodes.buttons);
+
+    this.#nodes.actions = make('div');
+
+    this.#nodes.holder.appendChild(this.#nodes.actions);
+
+    this.#eventBus.dispatchEvent(new InlineToolbarRenderedUIEvent({ toolbar: this.#nodes.holder }));
   }
 
   /**
-   * Attach all tools passed to the inline tool adapter
+   * Shows the Inline Toolbar
    */
-  #attachTools(): void {
-    Array.from(this.#tools.entries()).forEach(([toolName, tool]) => {
-      this.#formattingAdapter.attachTool(toolName as InlineToolName, tool.create());
-    });
+  #show(): void {
+    this.#nodes.holder.style.display = 'block';
   }
 
   /**
-   * Change toolbar show state if any text is selected
+   * Hides the Inline Toolbar
    */
-  #selectionChanged(): void {
-    /**
-     * Show or hide inline toolbar
-     */
-    if (this.#selectionRange !== undefined && this.#selectionRange[0] !== this.#selectionRange[1]) {
-      this.#createToolbarElement();
-    } else {
-      this.#deleteToolbarElement();
-    }
+  #hide(): void {
+    this.#nodes.holder.style.display = 'none';
   }
 
   /**
-   * @todo implement EventBus for ui to subscribe on selectionChange event
-   * Creates inline toolbar html element
+   * Moves the Inline Toolbar to the current selection
    */
-  #createToolbarElement(): void {
-    /**
-     * Before creating new toolbar element, remove existing one
-     */
-    this.#deleteToolbarElement();
+  #move(): void {
+    const selection = window.getSelection();
 
-    this.#toolbar = make('div');
+    if (!selection || !selection.rangeCount) {
+      return;
+    }
 
-    this.#tools.forEach((tool, toolName) => {
-      const inlineElementButton = make('button');
+    const range = selection.getRangeAt(0);
 
-      inlineElementButton.innerHTML = toolName;
+    const rect = range.getBoundingClientRect();
 
-      /**
-       * If tool has actions, then on click of the element button we should render actions element
-       * If tool has no action, then on click of the element button we should apply format
-       */
-      if (tool.hasActions) {
-        inlineElementButton.addEventListener('click', (_event) => {
-          this.#renderToolActions(createInlineToolName(toolName));
+    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+    this.#nodes.holder.style.top = `${rect.top + 16}px`;
+    this.#nodes.holder.style.left = `${rect.left}px`;
+    this.#nodes.holder.style.zIndex = '1000';
+  }
+
+  /**
+   * Renders the list of available inline tools in the Inline Toolbar
+   * @param tools - Inline Tools available for the current selection
+   */
+  #updateToolsList(tools: Map<InlineToolName, InlineTool>): void {
+    this.#nodes.buttons.innerHTML = '';
+
+    Array.from(tools.entries()).forEach(([name, tool]) => {
+      const button = make('button');
+
+      button.textContent = name;
+
+      if (Object.hasOwnProperty.call(tool.constructor.prototype, 'renderActions')) {
```

Comment:
there is ".hasActions()" in a facade

Reply:
Facade is not available here

Reply:
you can remove this method from the facade then